src/Charset.php

```diff
@@ -320,6 +320,23 @@ public function decodeCharset($encodedString, $charset)
         if ($charset == 'utf-8' || $charset == 'us-ascii') {
             return $encodedString;
         }
+        
+        // Handle gb2312 from Windows-based software products
+        if ($charset === 'gb2312') {
+            // Disable E_NOTICE errors to ignore E_NOTICE possibly triggered by iconv()
+            $currentErrorReporting = error_reporting();
+            error_reporting($currentErrorReporting & ~E_NOTICE);
+            
+            $result = iconv($charset, 'utf-8', $encodedString);
+            
+            error_reporting($currentErrorReporting);
```

Comment:
```suggestion
$result = iconv($charset, 'utf-8//ignore', $encodedString);
```
could be replace by this?

Reply:
No, because we don't want iconv to ignore non-convertible characters.
If we add //ignore, it returns a bad result.
By removing //ignore, iconv() will return false AND throw an E_NOTICE. The only way to ignore these E_NOTICE is to modify temporarily the error_reporting.

Reply:
If you really don't want to modify error_reporting, just remove lines that modify error_reporting.
This way, E_NOTICE could be thrown but it will be "normal".

Reply:
In my case, I have an app with many logs that include E_NOTICE and I don't want to have useless logs about these.

Reply:
Sorry I did a mistable my proposition was different in my head:

```suggestion
$result = iconv($charset, 'utf-8//translit', $encodedString);
```
or
```suggestion
$result = @iconv($charset, 'utf-8//translit', $encodedString);
```

Reply:
//translit will try to replace unknown characters by similar ones. This is not what we expect ^^

Reply:
And calling iconv with @ before will silent all php errors. We want to ignore E_NOTICE only ^^

Reply:
Thx for all the answers, I will review it during the holidays to merge it.